tests/foreman/api/test_product.py

```diff
@@ -64,46 +80,91 @@ def test_positive_create_with_gpg(self):
-            organization=org,
+            organization=self.org,
         ).create()
-        product = entities.Product(gpg_key=gpg_key, organization=org).create()
+        product = entities.Product(
+            gpg_key=gpg_key, organization=self.org).create()
         self.assertEqual(product.gpg_key.id, gpg_key.id)
 
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_name(self):
+        """Create a product providing invalid names only
 
-class ProductUpdateTestCase(APITestCase):
-    """Tests for updating a product."""
+        @Assert: A product is not created
 
-    @classmethod
-    def setUpClass(cls):  # noqa
-        """Create a product."""
-        super(ProductUpdateTestCase, cls).setUpClass()
-        cls.product = entities.Product().create()
+        @Feature: Product
+        """
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(HTTPError):
+                    entities.Product(name=name).create()
+
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_same_name(self):
+        """Create a product providing a name of already existent entity
+
+        @Assert: A product is not created
+
+        @Feature: Product
+        """
+        name = gen_string('alphanumeric')
+        entities.Product(name=name, organization=self.org).create()
+        with self.assertRaises(HTTPError):
+            entities.Product(name=name, organization=self.org).create()
 
+    @run_only_on('sat')
     @tier1
+    def test_negative_create_with_label(self):
+        """Create a product providing invalid label
+
+        @Assert: A product is not created
+
+        @Feature: Product
+        """
+        with self.assertRaises(HTTPError):
+            entities.Product(label=gen_string('utf8')).create()
+
     @run_only_on('sat')
-    def test_positive_update_name_and_description(self):
-        """Update a product with a new name or description.
+    @tier1
+    def test_positive_update_name(self):
+        """Update product name to another valid name.
 
-        @Assert: The given attributes are used.
+        @Assert: Product name can be updated.
 
         @Feature: Product
         """
-        for attrs in valid_name_desc_list():
-            with self.subTest(attrs):
-                setattr(self.product, attrs.keys()[0], attrs.values()[0])
-                self.product = self.product.update()
-                self.assertEqual(
-                    getattr(self.product, attrs.keys()[0]),
-                    attrs.values()[0]
-                )
+        product = entities.Product(organization=self.org).create()
+        for new_name in valid_data_list():
+            with self.subTest(new_name):
+                entities.Product(id=product.id, name=new_name).update(['name'])
+                updated = entities.Product(id=product.id).read()
+                self.assertEqual(new_name, updated.name)
```

Comment:
No need to create 2 new entities (and single-use variables) when you already have the required one:
```python
        product = entities.Product(organization=self.org).create()
        for new_name in valid_data_list():
            with self.subTest(new_name):
                product.name = new_name
                product = product.update(['name'])
                self.assertEqual(new_name, product.name)
```
Comment applies to other tests as well.